packages/map/src/polygon/index.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Layer, Source } from 'react-mapbox-gl';
+import { getFillPaint, getLinePaint, getTypeProp, getGeoJSON } from './utils';
+import { useChildrenBounds } from '../map/hooks';
+import { useID } from '../hooks';
+
+
+const Polygon = (props) => {
+  const {
+    type,
+    area,
+    onClick,
+  } = props;
+
+  useChildrenBounds(area);
+
+  const sourceId = useID();
+  const layerId = useID();
+
+  return (
+    <>
+      <Source geoJsonSource={getGeoJSON(area)} id={sourceId} />
+      <Layer id={layerId} type="fill" sourceId={sourceId} paint={getFillPaint(type)} onClick={onClick} />
+      <Layer type="line" sourceId={sourceId} paint={getLinePaint(type)} />
+    </>
+  );
+};
+
+Polygon.propTypes = {
+  type: getTypeProp().isRequired,
```

Comment:
i don't think utils here make sense, besides build strips this entire section out. if you use utils it's up to tree shake to determine unused utils (not sure if works)